downsample_rs/benches/bench_m4.rs

```diff
@@ -2,16 +2,15 @@ use downsample_rs::m4 as m4_mod;
 
 use criterion::{black_box, criterion_group, criterion_main, Criterion};
 use dev_utils::{config, utils};
-use ndarray::Array1;
 
 fn m4_f32_random_array_long_single_core(c: &mut Criterion) {
     let n = config::ARRAY_LENGTH_LONG;
     let data = utils::get_random_array::<f32>(n, f32::MIN, f32::MAX);
     c.bench_function("m4_scal_f32", |b| {
-        b.iter(|| m4_mod::m4_scalar_without_x(black_box(data.view()), black_box(2_000)))
+        b.iter(|| m4_mod::m4_without_x(black_box(data.as_slice()), black_box(2_000)))
     });
```

Comment:
Eventually we can remove all scalar benches (as this is now exactly the same code as the SIMD benches).
I would prefer to do this in a new PR (i.e., once this one is merged).